### PR TITLE
openssl3: Remove version holdback for OS <10.14.

### PR DESCRIPTION
**Pull request by fhgwright**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

```
Mac OS X 10.4.11 8S165, PPC, Xcode 2.5 8M2558
Mac OS X 10.4.11 8S2167, i386, Xcode 2.5 8M2558
Mac OS X 10.5.8 9L31a, PPC, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L31a, i386, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L31a, x86_64, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L31a, PPC (i386 Rosetta), Xcode 3.1.4 9M2809
Mac OS X 10.6.8 10K549, i386, Xcode 3.2.6 10M2518
Mac OS X 10.6.8 10K549, x86_64, Xcode 3.2.6 10M2518
Mac OS X 10.6.8 10K549, PPC (i386 Rosetta), Xcode 3.2.6 10M2518
Mac OS X 10.7.5 11G63, x86_64, Xcode 4.6.3 4H1503
OS X 10.8.5 12F2560, x86_64, Xcode 5.1.1 5B1008
OS X 10.9.5 13F1911, x86_64, Xcode 6.2 6C131e
OS X 10.10.5 14F2511, x86_64, Xcode 7.2 7C68
OS X 10.11.6 15G22010, x86_64, Xcode 8.1 8B62
macOS 10.12.6 16G2136, x86_64, Xcode 9.2 9C40b
macOS 10.13.6 17G14042, x86_64, Xcode 10.1 10B61
macOS 10.14.6 18G9323, x86_64, Xcode 11.3.1 11C505
macOS 10.15.7 19H15, x86_64, Xcode 12.4 12D4e
macOS 11.7.10 20G1427, x86_64, Xcode 13.2.1 13C100
macOS 11.7.10 20G1427, arm64, Xcode 13.2.1 13C100
macOS 12.7.5 21H1222, x86_64, Xcode 14.2 14C18
macOS 12.7.5 21H1222, arm64, Xcode 14.2 14C18
macOS 13.6.7 22G720, arm64, Xcode 15.2 15C500b
macOS 14.5 23F79, arm64, Xcode 15.4 15F31d
```

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: devel/openssl3/Portfile [hunk 1]
```diff
@@ -53,26 +53,23 @@ checksums           rmd160  f7d3736a023dcf7d40db016182ca6d1de5a6fa69 \
                     sha256  777cd596284c883375a2a7a11bf5d2786fc5413255efab20c50d6ffe6d020b7e \
                     size    18055752
 
-# 3.2.0 is currently broken for OS < 10.14, see #68766
-if {${os.platform} eq "darwin" && ${os.major} < 18} {
-
-    subport ${name}-devel {
-        conflicts           ${name}
-    }
```
Comment (neverpanic) [inline devel/openssl3/Portfile:61]: Strictly speaking this silently drops the `openssl-devel` port, but I'm fine with that.

**Review by neverpanic — APPROVED** [review]

**Comment by neverpanic** [comment]
Thanks for doing the testing work for this!

File: devel/openssl3/Portfile (hunk 1 above)
Comment (fhgwright) [reply devel/openssl3/Portfile:61]: Ah, you're right.  I guess strictly speaking, it should have been obsoleted in favor of the main port.  But probably the only users were a few people (besides myself) testing the new version on the older platforms, so it's not clear it's worth worrying about.

Comment (fhgwright) [reply devel/openssl3/Portfile:61]: Actually, properly obsoleting this is more important than appears at first glance, due to the deficient handling of existing installs that don't match current Portfiles.  Although each installed instance keeps a copy of the Portfile used to create it, base does a poor job of using that information, leading to at least two known problems:

1. Activating a downlevel port instance apparently computes its dependencies based on its _current_ Portfile, not the saved one.
2. Attempts to clean an installed port with no current Portfile are refused, based on the lack of a current PortIndex entry.

Thus, I think it's worth properly obsoleting the `opensl3-devel` port.  PR coming.

**Pull request merged**